docs(launch-autoware): add how to launch yabloc or eagleye (#411)
autowarefoundation/autoware-documentation: docs/how-to-guides/integrating-autoware/launch-autoware/localization-mode/yabloc-guide.md

> # Using YabLoc in Autoware

Reviewer: It's rather a broad suggestion which is pretty much open to discussion, but since it seems that the contents are mostly duplicated with the Readme.md of YabLoc in Autoware Universe. I guess it's better to avoid maintaining such duplicated documents, so how about removing this document and merge it into Readme, or the other way around (Remove readme and just put a link to this document)?

Author: Thank you for the suggestion. It's a great idea.
Since the "limitations" and "input/output" are already written in the README of `autoware.universe`, I will try to reconsider the content of this page to avoid duplication. :raising_hand_man:

## What is YabLoc?

YabLoc is a camera-based localization method.
It estimates position by matching road surface markings extracted from images with a vector map.
Point cloud maps and LiDAR are not required.

> YabLoc enables users localize with vehicles that are not equipped with LiDAR and in environments where point cloud maps are not available.

Reviewer: ```suggestion
YabLoc enables users localize vehicles that are not equipped with LiDAR and in environments where point cloud maps are not available.
```

### Principle

> The diagram below illustrates the basic principle of YabLoc. It extracts road surface markings by masking the line segments using the road area obtained from graph-based segmentation. The red line at the center-top of the diagram represents the line segments identified as road surface markings. YabLoc transforms these segments for each particle and determines the particle's weight by comparing them with the cost map generated from Lanelet2.

Reviewer: ```suggestion
The diagram below illustrates the basic principle of YabLoc. It extracts road surface markings by extracting the line segments using the road area obtained from graph-based segmentation. The red line at the center-top of the diagram represents the line segments identified as road surface markings. YabLoc transforms these segments for each particle and determines the particle's weight by comparing them with the cost map generated from Lanelet2.
```

![node_diagram](images/yabloc-integration-guide/yabloc_principle.png)

### YabLoc Packages

- [yabloc_common](https://github.com/autowarefoundation/autoware.universe/tree/main/localization/yabloc/yabloc_common/README.md)
- [yabloc_image_processing](https://github.com/autowarefoundation/autoware.universe/tree/main/localization/yabloc/yabloc_image_processing/README.md)
- [yabloc_particle_filter](https://github.com/autowarefoundation/autoware.universe/tree/main/localization/yabloc/yabloc_particle_filter/README.md)
- [yabloc_pose_initializer](https://github.com/autowarefoundation/autoware.universe/tree/main/localization/yabloc/yabloc_pose_initializer/README.md)

## How to Launch YabLoc

When launching autoware, if you set `localization_mode:=camera` as an argument, YabLoc will be launched instead of NDT.
By default, `localization_mode` is `lidar`.

A sample command to use YabLoc is as follows

```shell
ros2 launch autoware_launch logging_simulator.launch.xml \
  map_path:=$HOME/autoware_map/sample-map-rosbag\
  vehicle_model:=sample_vehicle \
  sensor_model:=sample_sensor_kit \

>   localization_mode:=yabloc

Reviewer: ```suggestion
  localization_mode:=camera
```

| ----- | ----------------------------------------------------------------------- | ----------------------------------------------------------------------------- |
| 1     | `/localization/yabloc/pf/cost_map_image`                                | cost map made from lanelet2                                                   |
| 2     | `/localization/yabloc/pf/match_image`                                   | projected line segments                                                       |
| 3     | `/localization/yabloc/image_processing/image_with_colored_line_segment` | classified line segments. green line segments are used in particle correction |
| 4     | `/localization/yabloc/image_processing/lanelet2_overlay_image`          | overlay of lanelet2                                                           |
| 5     | `/localization/yabloc/image_processing/segmented_image`                 | graph based segmentation result                                               |

## Limitation

- **Running YabLoc and NDT simultaneously is not supported.**

>   - This is because both are computationally expensive.

Reviewer: Maybe this is what you meant?
```suggestion
  - This is because running both at the same time may be computationally too expensive.
```

  - Also, in most cases, NDT is superior to YabLoc, so there is less benefit to running them at the same time.
- It does not estimate roll and pitch, therefore some of the perception nodes may not work well.
- It doesn't support multiple cameras now. But it will in the future.

> - In places where there are few road surface markings, such as intersections, the estimation heavily relies on GNSS, IMU, and vehicles twist.

Reviewer: ```suggestion
- In places where there are few road surface markings, such as intersections, the estimation heavily relies on GNSS, IMU, and vehicle's wheel odometry.
```

> - If the road boundary or road surface markings are not included in the Lanelet2, the estimation fails.

Reviewer: ```suggestion
- If the road boundary or road surface markings are not included in the Lanelet2, the estimation is likely to fail.
```
